sdk/dotnet/Pulumi.Automation.Tests/RemoteWorkspaceTests.cs

```diff
@@ -1,11 +1,204 @@
+                {
+                    Args = new RemoteGitProgramArgs("owner/project/stack/wat", _testRepo),
+                    Error = "StackName \"owner/project/stack/wat\" not fully qualified.",
+                },
+                new
+                {
+                    Args = new RemoteGitProgramArgs(stackName, null!),
+                    Error = "Url is required.",
+                },
+                new
+                {
+                    Args = new RemoteGitProgramArgs(stackName, ""),
+                    Error = "Url is required.",
+                },
+                new
+                {
+                    Args = new RemoteGitProgramArgs(stackName, _testRepo),
+                    Error = "either Branch or CommitHash is required.",
+                },
+                new
+                {
+                    Args = new RemoteGitProgramArgs(stackName, _testRepo)
+                    {
+                        Branch = "",
+                        CommitHash = "",
+                    },
+                    Error = "either Branch or CommitHash is required.",
+                },
+                new
+                {
+                    Args = new RemoteGitProgramArgs(stackName, _testRepo)
+                    {
+                        Branch = "branch",
+                        CommitHash = "commit",
+                    },
+                    Error = "Branch and CommitHash cannot both be specified.",
+                },
+                new
+                {
+                    Args = new RemoteGitProgramArgs(stackName, _testRepo)
+                    {
+                        Branch = "branch",
+                        Auth = new RemoteGitAuthArgs
+                        {
+                            SshPrivateKey = "key",
+                            SshPrivateKeyPath = "path",
+                        },
+                    },
+                    Error = "SshPrivateKey and SshPrivateKeyPath cannot both be specified.",
+                },
+            };
+
+            foreach (var factory in factories)
+            {
+                foreach (var test in tests)
+                {
+                    yield return new object[] { factory, test.Args, test.Error };
+                }
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(ErrorsData))]
+        public async Task Errors(Func<RemoteGitProgramArgs, Task<RemoteWorkspaceStack>> factory,
+            RemoteGitProgramArgs args, string error)
+        {
+            var exception = await Assert.ThrowsAsync<ArgumentException>(() => factory(args));
+            Assert.Equal(error, exception.Message);
+        }
+
+        [ServiceRequiredFact]
+        public Task CreateStackLifecycle()
+            => TestStackLifeCycle(RemoteWorkspace.CreateStackAsync);
+
+        [ServiceRequiredFact]
+        public Task CreateOrSelectStackLifecycle()
+            => TestStackLifeCycle(RemoteWorkspace.CreateOrSelectStackAsync);
```

Comment:
Unfortunately, these each take ~3 minutes. What can we do to avoid running them in PRs?

Reply:
We'll disable the test by default (requires `PULUMI_TEST_DEPLOYMENTS_API` to be set). And consider running the test in a cron until we add support for mocking the deployments api.